### PR TITLE
Signup: Update site-or-domain step

### LINKED ISSUE #12635: Not clear your domain is included with a plan in the /domains flow
In the /domains flow, we could make it clearer that the domain you just chose is included with the plan. This is a selling point that we're not capitalizing on.

From @sararosso (referring to the plans page):
>maybe we should bold / replace this text with “Your custom domain name is included!” or maybe even the name of the domain?

@danhauk worked on a design to highlight this in the results:

image

### PR DESCRIPTION
**Pull request by scruffian**
This pull request fixes #12635 by updating the `site-or-domain` step with new domains, and updating the tooltip for custom domains on the plans page to show the selected domain.

screen shot 2017-04-26 at 11 25 35
screen shot 2017-04-26 at 11 25 24

This also updates the mobile view:
screen shot 2017-04-28 at 14 27 52

#### Testing instructions
  
1. Run `git checkoutupdate/site-or-domain-step` and start your server, or open a [live branch](https://delphin.live/?branch=update/site-or-domain-step)
2. Open the [`Domain first signup` page](http://calypso.localhost:3000/start/domain-first/site-or-domain?new=adfgadfgd23453.com)
3. Check that you see the new design for this screen
2. Open the [`Regular signup` page](http://calypso.localhost:3000/start)
2. Select a domain name
3. Check that you see the domain name in the tooltip on the plans page
  
#### Reviews
  
- [x] Code
- [ ] Product

@Automattic/sdev-feed

### CODE REVIEW

File: client/signup/steps/design-type-with-store/new-site-image.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+				<use xlinkHref="#path-1"></use>
+			</mask>
+		</defs>
+		<g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+			<g id="Group-3-Copy">
+				<g id="Group-2">
+					<g id="domain-illustration">
+						<g id="Imported-Layers">
+							<path d="M0.102747909,15.6269981 L0.102747909,2.72646653 C0.102747909,1.30182267 1.26010036,0.147016357 2.6878853,0.147016357 L169.225395,0.147016357 C170.653386,0.147016357 171.810944,1.30223275 171.810944,2.72708166 L171.810944,15.6269981 L0.102747909,15.6269981" id="Fill-1" fill="#CFDEE7"></path>
+							<use id="Fill-6" stroke="#E6F0F6" mask="url(#mask-2)" stroke-width="2" fill="#FFFFFF" xlinkHref="#path-1"></use>
```
Comment (danhauk) [inline client/signup/steps/design-type-with-store/new-site-image.jsx]: `stroke` should be `#D9E8F0` to darken the illustration stroke and match the domain-only illustration.

**Comment by danhauk** [comment]
Tested for logged out and logged in flow. Working as expected.

There is a slight inconsistency between the borders of the illustrations - the new site illustration has a lighter border than the domain only one. We should darken the new site one to provide some more contrast and consistency. I made a comment at the appropriate line which should fix it.

Otherwise product LGTM 👍

File: client/my-sites/plan-features/index.jsx [hunk 2]
```diff
@@ -331,7 +331,7 @@ class PlanFeatures extends Component {
 
 	renderFeatureItem( feature, index ) {
 		const description = feature.getDescription
-					? feature.getDescription( abtest )
+					? feature.getDescription( abtest, this.props.domainName )
```
Comment (aidvu) [inline client/my-sites/plan-features/index.jsx]: We prolly don't need the `abtest` var anymore?

File: client/signup/steps/site-or-domain/index.jsx [hunk 3]
```diff
@@ -67,12 +70,21 @@ export default class SiteOrDomain extends Component {
 		return (
 			<div className="site-or-domain__choices">
 				{ this.getChoices().map( ( choice ) => (
-					<Card className="site-or-domain__choice" key={ choice.type }>
-						<a href="#" onClick={ ( event ) => this.handleClickChoice( event, choice.type ) }>
-							{ choice.image }
-							<h2>{ choice.label }</h2>
-						</a>
-					</Card>
+					<div className="site-or-domain__choice" key={ choice.type }>
```
Comment (aidvu) [inline client/signup/steps/site-or-domain/index.jsx]: I'd either make the whole boxes clickable, or just the button. It's a bit confusing that you can click the image/button but not outside those two.

**Pull request merged**